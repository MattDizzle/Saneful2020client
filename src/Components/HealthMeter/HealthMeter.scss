// .barContainer {
//   height: 100%;
//   width: 100%;
//   background-color: #e0e0de;
//   margin: 5px 0;
//   border: 5px solid darkgray;
//   border-style: ridge;
//   border-radius: 11px;
// }

// .barLabel {
//   padding: 5;
//   color: white;
// }