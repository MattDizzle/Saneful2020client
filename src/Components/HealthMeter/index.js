import React from "react";

import './HealthMeter.scss';

const HealthMeter = (props) => {
  const { currentHealth } = props;

  const fillerStyles = {
    height: '100%',
    borderRadius: '10px',
    transition: '0.25s',
    width: `${currentHealth}%`,
    backgroundColor: '#ab111b',
    textAlign: 'right'
  };

  return (
    //entire bar
    //completed percentage
    <div className="title-barContainer">
      <p>Health:</p>
      <div className="barContainer">
        <div style={fillerStyles}>
          <span className="barLabel">{`${currentHealth}`}</span>
        </div>
      </div>
    </div>

  );
};

export default HealthMeter;