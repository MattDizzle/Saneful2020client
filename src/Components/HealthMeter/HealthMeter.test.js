import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import HealthMeter from './index';


describe(`HealthMeter Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <HealthMeter />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <HealthMeter />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})