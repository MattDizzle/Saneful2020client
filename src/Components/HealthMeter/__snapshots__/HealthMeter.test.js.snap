// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthMeter Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="title-barContainer"
>
  <p>
    Health:
  </p>
  <div
    className="barContainer"
  >
    <div
      style={
        Object {
          "backgroundColor": "#ab111b",
          "borderRadius": "10px",
          "height": "100%",
          "textAlign": "right",
          "transition": "0.25s",
          "width": "undefined%",
        }
      }
    >
      <span
        className="barLabel"
      >
        undefined
      </span>
    </div>
  </div>
</div>
`;
